sys/dev/drm/core/drm_file.c

```diff
@@ -603,6 +608,52 @@ ssize_t drm_read(struct file *filp, char __user *buffer,
 				return ret;
 		} else {
 			unsigned length = e->event->length;
+			void *ptr;
+
+			ptr = e->event;
+
+#ifdef COMPAT_FREEBSD64
+			struct drm_event_crtc_sequence64 crtc_seq64;
+			struct drm_event_crtc_sequence *crtc_seq;
+			struct drm_event_vblank64 vbl64;
+			struct drm_event_vblank *vbl;
+
+			if (SV_CURPROC_FLAG(SV_CHERI))
+				goto done;
+
+			switch(e->event->type) {
```

Comment:
Also this feels like it should be _after_ the if below, just before the copy_to_user?